server/plugin.go

```diff
@@ -81,7 +84,9 @@ func (p *Plugin) OnActivate() error {
 	p.makeChannelPostsIterator = func(channel *model.Channel, showEmailAddress bool) PostIterator {
 		return channelPostsIterator(p.client, channel, showEmailAddress)
 	}
-	return registerAPI(p.router, p.client, p.makeChannelPostsIterator)
+	p.handler = Handler{}
```

Comment:
Could it be better to have `p.registerAPI` return the handler that goes in there?